### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by jmzwar**
Dashboard First version
Screen Shot 2023-04-23 at 7 21 29 pm

### CODE REVIEW

File: v2/perps-v2/ui/src/hooks/useActions.ts [hunk 1]
```diff
@@ -162,7 +167,7 @@ export const useActions = (account?: string) => {
 
   return {
     loading: marginLoading || futuresTradesLoading,
-    data: sortedData,
+    data: limit ? sortedData.slice(0, limit) : sortedData,
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useActions.ts:170]: Why was this needed?

File: v2/perps-v2/ui/src/hooks/useTrades.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+function calculateMonths() {
+  const months: TradesRange[] = [];
+
+  for (let i = 0; i <= 12; i++) {
+    const day = subMonths(new Date(), 12 - i);
+    const timestamp = getUnixTime(day);
+    const label = format(day, 'dd/MM');
+    months.push({ timestamp, label, trades: new Set(), count: 0 });
+  }
+  return months;
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useTrades.ts:43]: I think we need to move this logic to the subgraph. I think current logic would behave differently based on timezone and is a bit slow

File: v2/perps-v2/ui/src/hooks/useTrades.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+): Promise<{ hydratedRange: TradesRange }> {
+  try {
+    const { data } = await client.query({
+      query: TRADES_QUERY,
+      variables: {
+        where: {
+          timestamp_gte: `${lowerTimestamp}`,
+          timestamp_lte: `${upperTimestamp}`,
+        },
+        skip,
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useTrades.ts:63]: This will fail if we have more than 5k trades :( Subgraph only allows you to skip 5k items.

File: v2/perps-v2/ui/src/hooks/useTraders.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      // Otherwise they are returning.
+      const nextTimestamp = timeRange[index + 1]?.timestamp;
+      const firstTrade = parseInt(trader.timestamp);
+
+      if (
+        typeof nextTimestamp === 'undefined' &&
+        timestamp < firstTrade &&
+        firstTrade < Math.floor(Date.now() / 1000)
+      ) {
+        newTraders++;
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useTraders.ts:139]: I do wonder if this logic is correct, when I saw the "new traders" I would assume the design meant new trader to perps, not new for this month.  Which means we need to move this logic to subgraph too, unless we want to fetch all trades since deployment

File: v2/perps-v2/ui/src/components/Actions/Dashboard/LargestLosses.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import { POSITIONS_QUERY_MARKET } from '../../../queries/positions';
+import { Market, PnL, TableHeaderCell, WalletAddress } from '../../Shared';
+import { FuturesPosition_OrderBy, OrderDirection } from '../../../__generated__/graphql';
+import { SmallTableLoading } from './SmallTableLoading';
+
+export const LargestLosses = () => {
+  const { data, loading, error } = useQuery(POSITIONS_QUERY_MARKET, {
+    variables: {
+      where: {
+        openTimestamp_gte: `${getUnixTime(subDays(new Date(), 1))}`,
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/components/Actions/Dashboard/LargestLosses.tsx:14]: hmm I'm not sure this is totally right? Here we're only filter on position opened in the last day.  I would have thought that we want to include a position that was opened a week ago and closed today?

We might want to use trades instead of position here

File: v2/perps-v2/ui/src/hooks/useActions.ts (hunk 1 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useActions.ts:170]: Can't remember but I've removed it

File: v2/perps-v2/ui/src/queries/resolved.ts [hunk 6]
```diff
@@ -65,5 +65,10 @@ export const resolvers: Resolvers | Resolvers[] = {
 
         .filter(notNill);
     },
+    marketsData: async (_parent, _args, _contextValue, _info) => {
+      return {
+        hello: 'world',
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/queries/resolved.ts:70]: haha?

Comment (jmzwar) [reply v2/perps-v2/ui/src/queries/resolved.ts:70]: ooh

File: v2/perps-v2/ui/src/hooks/useMarketStats.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+export function useMarketStats() {
+  const { upper, lower } = getDateRange(1, 2); // 1 day ago, 2 days ago UTC
+
+  const { data: marketsData, loading } = useQuery(MARKETS_QUERY, {
+    variables: {
+      first: 3,
+      orderBy: DailyMarketStat_OrderBy.Timestamp,
+      orderDirection: OrderDirection.Desc,
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useMarketStats.ts:23]: Ah damn you are right. We need to be order by Volume

File: v2/perps-v2/ui/src/components/Header.tsx [hunk 8]
```diff
@@ -24,26 +35,87 @@ export const Header: FC = () => {
...
       borderBottomWidth="1px"
       borderBottomColor="gray.900"
+      borderTopWidth="1px"
+      borderTopColor="transparent"
     >
       <Flex alignItems="center">
         <RouterLink to="/">
           <SNXIcon />
-          <PerpsStats mt="3px" ml={3} />
+          <PerpsStats mt="3px" ml={3} display={{ base: 'none', c900: 'initial' }} />
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply v2/perps-v2/ui/src/components/Header.tsx:44]: Custom breakpoint
